test/test_test.go

```diff
@@ -66,22 +70,44 @@ func newSlowProxy(latency time.Duration, opts *server.Options) (*slowProxy, *ser
 			panic("Can't connect to server")
 		}
 		sp.conns = append(sp.conns, client, server)
-		go sp.loop(latency, client, server)
-		go sp.loop(latency, server, client)
+		go sp.loop(rtt, up, client, server)
+		go sp.loop(rtt, down, server, client)
 	}()
-	sopts := &server.Options{Host: "127.0.0.1", Port: port}
-	return sp, sopts
+	sp.opts = &server.Options{Host: "127.0.0.1", Port: port}
+	sp.u = fmt.Sprintf("nats://%s:%d", sp.opts.Host, sp.opts.Port)
+	return sp
 }
 
-func (sp *slowProxy) loop(latency time.Duration, r, w net.Conn) {
-	delay := latency / 2
-	for {
-		var buf [1024]byte
+func (sp *slowProxy) Opts() *server.Options {
```

Comment:
minor but since slowProxy is not exported, there is no need to export Opts() and ClientURL(). You can leave as functions, of course, but no need to have upper case. (that being said, since sp.opts exists you would need to rename that to sp.o if you want to have sp.opts()).

Reply:
I think we will expose eventually but can make the functions un-exported for now. Will fix.